src/websocket-service.ts

```diff
@@ -1,48 +1,60 @@
+import express from 'express'
+import passport from 'passport'
+import LocalStrategy from 'passport-local'
+import cookieParser from 'cookie-parser'
+import bodyParser from 'body-parser'
+import session from 'express-session'
 import ws from 'ws'
 import http from 'http'
 import setupWSConnection from './socket-message-handling';
 
-/**
- *
- * [`y-websocket`](https://github.com/yjs/y-websocket) provides a template
- * websocket services to interface with their websocket provider
- * https://github.com/yjs/y-websocket/blob/master/bin/server.js
- * 
- * [`y-webrtc`](https://github.com/yjs/y-webrtc) provides
- * a thin signalling backing service that connects to their
- * webrtc client provider https://github.com/yjs/y-webrtc/blob/master/bin/server.js 
- *
- * TODO: Build this. 
- * 
- * Functions from both services should be combined here, to produce
- * a single websocket connection that provides both websocket
- * push changes, and initial document loads, as well as connecting
- * peers to each other.
- * 
- * Data persistance.
- *   The y-websocket interface implements a leveldb data layer interface.
- *   * What is leveldb?
- *   * Implement connector to some docker database.
- */
- 
-
 const host = process.env.HOST || '0.0.0.0'
 const port = process.env.PORT || 6001;
-// @ts-ignore
-const wss = new ws.Server({ noServer: true });
+const SESSION_SECRET = process.env.SESSION_SECRET || 'big secret'
+
+const app = express();
```

Comment:
Should the express app be in a separate file?

Reply:
🤷  depends on how complicated it gets. Might be worth having a cleaner separation between backend and frontend code though. Different packages with yarn workspaces might be a nice pattern here.